Library/OcBootManagementLib/OcBootManagementLib.c

```diff
@@ -525,26 +536,49 @@ OcRunFirmwareApplication (
 }
 
 EFI_STATUS
-OcLaunchAppleBootPicker (
-  VOID
+OcGetAppleFirmwareUI (
+  IN APPLE_FIRMWARE_USER_INTERFACE_PROTOCOL  **UIProtocol,
+  IN APPLE_FIRMWARE_UI_VARS                  **UIVars OPTIONAL
   )
 {
-  EFI_STATUS                              Status;
-  APPLE_FIRMWARE_USER_INTERFACE_PROTOCOL  *FirmwareUI;
-  UINT8                                   *aGopAlreadyConnected;
+  EFI_STATUS  Status;
+
+  ASSERT (UIProtocol != NULL);
 
   Status = gBS->LocateProtocol (
                   &gAppleFirmwareUserInterfaceProtocolGuid,
                   NULL,
-                  (VOID **)&FirmwareUI
+                  (VOID **)UIProtocol
                   );
 
+  if (EFI_ERROR (Status)) {
+    return Status;
+  }
+
+  if (UIVars != NULL) {
+    *UIVars = (VOID *)((UINT8 *)*UIProtocol + sizeof (APPLE_FIRMWARE_USER_INTERFACE_PROTOCOL));
```

Comment:
Imo write a wrapper structure with the protocol as first member and then use `BASE_CR`

Reply:
Done in commit to follow.